utils/mil_tools/scripts/pythonFinder.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python
+import os
+import re
+import sys
+import subprocess
+import shlex
+
+# Joseph Brooks
+# Checks NaviGator for all python files
+
+if len(sys.argv) == 1:
+    print("add directory path to be searched as an argument")
+    sys.exit()
+
+directory = sys.argv[1]
+
+allFileList = []
+
+# gets all files in directory
+for root, dirs, files in os.walk(directory):
+    for name in files:
+        programName = os.path.join(root, name)
+        allFileList.append(programName)
+
+pythonList = []
+
+# attempts to open file as text and checks #!/usr/bin/env
+# Just checks the first line
+for program in allFileList:
+    try:
+        for line in open(program):
+            if re.search("#!/usr/bin/env python", line):
+                pythonList.append(program)
+            break
+    except:
+        pass
+
+# checks for ending .py
+for program in allFileList:
+    if '.py' in program and '.pyc' not in program:
+        pythonList.append(program)
+
+# checking for duplicates
+finalList = []
+
+for program in pythonList:
+    # do not add if duplicate
+    if program in finalList:
+        pass
+    else:
+        finalList.append(program)
+
+counter = 0
+for program in finalList:
```

Comment:
This should ideally be just ONE command which runs, checking all files by adding each as an argument.